viewing_party/party.py

```diff
@@ -1,23 +1,132 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating: 
+        return { "title": title, "genre": genre, "rating": rating}
+    else:
+        return None
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    watched = user_data["watched"]
+    for movie in watchlist:
+        if movie["title"] == title:
+            watchlist.remove(movie)
+            watched.append(movie)
+            return user_data
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_sum = 0
+    if len(user_data["watched"]) < 1:
+        return 0.0
+    for movie in user_data["watched"]:
+        rating_sum += movie["rating"]
+        rating_sum = float(rating_sum)

```

Comment:
This line isn't necessary! Once we do the division on line 41, the number will become a float. We do not need to manually turn it into a float.